Add a norm to the EBL model

JouvinLea opened the pull request:
This PR adds a norm to the EBL model to give the possibility to the user to add a norm for the AbsorbedSpectralModel class and also to let it free in the fits.

It is also adding a small argument for the peek() method of FLuxPointDataset because for now the uncertainties are not handle properly and it crashed if you try to plot the error of this model.

cdeil commented on gammapy/modeling/models/spectral.py:
```diff
@@ -1422,13 +1422,23 @@ class AbsorbedSpectralModel(SpectralModel):
         parameter name
+    alpha_norm: float
+        Norm of the EBL model, by default 1.
+    alpha_norm_frozen: bool
```
@JouvinLea - Please remove this `alpha_norm_frozen` option.
Just put `frozen=True` in the `Parameter` call in `__init__` below.
If someone wants to change this, they can change to `model.alpha_norm.frozen = False` after `__init__`.
This is what we do for all other cases where parameters are usually frozen, e.g. the reference energies in spectral models.
We will improve the model init methods, but manually adding options like `alpha_norm_frozen` is not the way to go.

cdeil commented on gammapy/modeling/models/spectral.py:
```diff
@@ -1440,6 +1450,8 @@ def __init__(
         parameters = spectral_model.parameters.parameters.copy()
         parameters.append(par)
+        self.alpha_norm = Parameter("alpha_norm", alpha_norm, frozen=alpha_norm_frozen)
+        parameters.append(self.alpha_norm)
```
Use this instead of two append calls:
```
parameters.extend([par, self.alpha_norm])
```
I think this works too and does the same thing, you can use that if you prefer:
```
parameters += [par, self.alpha_norm]
```

cdeil commented on gammapy/modeling/tests/test_serialize_yaml.py:
```diff
@@ -130,8 +130,8 @@ def test_datasets_to_io(tmpdir):
-        dataset0.model.skymodels[0].parameters["reference"]
-        is dataset1.model.skymodels[1].parameters["reference"]
+            dataset0.model.skymodels[0].parameters["reference"]
+            is dataset1.model.skymodels[1].parameters["reference"]
```
Please revert this extra indentation. We use 4 space indentation, you changed to 8 space indentation here for some reason.

cdeil commented on gammapy/spectrum/flux_point.py:
```diff
@@ -1289,21 +1289,23 @@ def residuals(self, method="diff"):
         method : {"diff", "diff/model", "diff/sqrt(model)"}
             Method used to compute the residuals, see `MapDataset.residuals()`
+        plot_error : bool
+            False to not plot the errors of the model since for some of them it is not handle yet properly in Gammapy
```
Please shorten description to:
```
Plot error band?
```

cdeil commented on gammapy/spectrum/flux_point.py:
```diff
@@ -1393,6 +1397,8 @@ def plot_spectrum(self, ax=None, fp_kwargs=None, model_kwargs=None):
         model_kwargs : dict
             Keywords passed to `SpectralModel.plot` and `SpectralModel.plot_error`
+        plot_error : bool
+            False to not plot the errors of the model since for some of them it is not handle yet properly in Gammapy
```
Please shorten description (see comment above)

JouvinLea commented:
@cdeil 
I fixed the small docs and space issues.
I added a description for the Adsorbed model, hope it is clearer!

Personally I would prefer to stay with this plot_error parameter. I put a TODO to say to remove it when the issue is fixed. I know exactly which line to comment to have my script running but this is a useful plot routine so I think for non developper it is better to have this option. But as you wish, I can comments those lines by myself each time I run my script.

adonath commented on gammapy/spectrum/flux_point.py:
```diff
@@ -1426,7 +1432,8 @@ def plot_spectrum(self, ax=None, fp_kwargs=None, model_kwargs=None):
 
-        if self.model.parameters.covariance is not None:
+        # TODO: remove this plot error option when there is a better manipulation of uncertainties
+        if self.model.parameters.covariance is not None and plot_error:
```
@JouvinLea Maybe just go with a try except statement here and remove the `plot_error` option? Like so:

```
#TODO: remove try except....
try:
    self.model.plot_error()
except WhatEverError:
    log.debug("Model does not support evaluation of errors")
    pass

```

cdeil commented:
@JouvinLea - Please fix:
https://dev.azure.com/gammapy/gammapy/_build/results?buildId=2035&view=logs&jobId=d7d66435-5ae4-561a-5876-2cd4e19b3ce2&taskId=922dfe32-5b90-53bc-18db-0fa0ecbbd97b&lineStart=187&lineEnd=188&colStart=1&colEnd=1

cdeil commented:
@JouvinLea - Thanks!
